File: Practice/15/C++/guess_the_number.cpp
#include <iostream>

/* Задание 15 - Угадай число */

using namespace std;

bool restart()
{
    int a;
    cout << "Хотите начать сначала? (1 - ДА ): ";
    cin >> a;
    if (a == 1)
        return true;
    else
        return false;
}

int main()
{
    cout << "15. Угадай число" << endl;
    int n, cnum;

    while (true)
    {
        int attempts = 5;

        srand(unsigned(time(NULL)));
        cnum = rand() % 100 + 1;

        for (int i = 0; i != attempts; i++)
        {
            cout << "Введите число: ";
            cin >> n;

            if (n == cnum)
            {
                cout << "Поздравляю! Вы угадали" << endl;
                if (restart())
                    continue;
                else
                    break;
            }
            else if (n < cnum)
            {
                cout << "Загаданное число больше" << endl;
            }
            else if (n > cnum)
            {
                cout << "Загаданное число меньше" << endl;
            }
        }
        if (attempts <= 0)
            cout << "Вы проиграли. Было загадано: " << cnum << endl;
        if (restart())
            continue;
        else
            break;
    }
    return 0;
}

File: Practice/15/Python/guess_the_number.py
#!/usr/bin/python3
from random import randrange

# Задание 15 - Угадай число


def restart():
    if (int(input("Хотите начать сначала? (1 - ДА ): ")) == 1):
        return True
    else:
        return False


def game():
    attempts = 5
    cnum = randrange(101)

    for i in range(0, attempts):
        n = int(input("Введите число: "))

        if (n == cnum):
            print("Поздравляю! Вы угадали")
            return
        elif (n < cnum):
            print("Загаданное число больше")
        elif (n > cnum):
            print("Загаданное число меньше")

    print("Вы проиграли. Было загадано: {}".format(cnum))
    return


def main():
    print("15. Угадай число")
    inGame = True
    while inGame:
        game()
        inGame = restart()


if __name__ == "__main__":
    main()
